spec/controllers/api/v1/offers_controller_spec.rb

```diff
@@ -86,6 +87,35 @@
         get :index, created_by_id: offer1.created_by_id
         expect(assigns(:offers).to_a).to eql([offer1])
       end
+
+      describe "if user is supervisor" do
+        context "for donor app"
+          let!(:offer1) { create :offer, created_by: supervisor }
+          let!(:offer2) { create :offer }
+          before { generate_and_set_token(supervisor) }
+
+          it 'returns offers created_by supervisor' do
+            request.headers["X-GOODCITY-APP-NAME"] = "app.goodcity"
+            get :index
+            expect(assigns(:offers).to_a).to eql([offer1])
+            expect(assigns(:offers).to_a).not_to include(offer2)
```

Comment:
technically the line above does the same check

Reply:
Yup.. Thanks @steveyken . I will remove that unnecessary check. :)